Utilities/IntervalFiles/ComputeIntervalBamStats.wdl

```diff
@@ -0,0 +1,306 @@
+    }
+
+    output {
+        File bam_subset = "~{interval_name}.bam"
+        File bam_subset_index = "~{interval_name}.bai"
+    }
+}
+
+# For coverage-related stats and distribution
+task ComputeWgsMetrics {
+    input {
+        File input_bam
+        File input_bam_index
+
+        File ref_fasta
+        File ref_index
+        File ref_dict
+
+        File interval_file
+        String interval_name
+
+        # Tool arguments
+        Int min_BQ = 20
+        Int min_MQ = 20
+        Int coverage_cap = 250
+
+        # Runtime
+        String gatk_tag = "4.3.0.0"
+        Int disk_size = 2 * ceil(size(input_bam, "GB")) + 25
+        Int cpu = 8
+        Int memory = 32
+    }
+
+    command <<<
+        set -xueo pipefail
+
+        gatk CollectWgsMetrics \
+            -I ~{input_bam} \
+            -R ~{ref_fasta} \
+            -O "~{interval_name}.wgs_metrics" \
+            --INTERVALS ~{interval_file} \
+            ~{"--MINIMUM_BASE_QUALITY " + min_BQ} \
+            ~{"--MINIMUM_MAPPING_QUALITY " + min_MQ} \
+            ~{"--COVERAGE_CAP " + coverage_cap}
+
+    >>>
+
+    runtime {
+        docker: "us.gcr.io/broad-gatk/gatk:" + gatk_tag
+        disks: "local-disk " + disk_size + " HDD"
+        cpu: cpu
+        memory: memory + " GB"
+    }
+
+    output {
+        File wgs_file = "~{interval_name}.wgs_metrics"
+    }
+}
+
+task ComputeMAPQDistribution {
+    input {
+        File input_bam
+        File input_bam_index
+
+        File interval_file
+        String interval_name
+
+        # Runtime
+        Int cpu = 8
+        Int memory = 16
+        Int extra_disk = 25
+
+    }
+
+    Int disk_size = 2*ceil(size(input_bam, "GB")) + extra_disk
+
+    command <<<
+        set -xueo pipefail
+
+        samtools view ~{input_bam} | awk '{print $5}' | sort -n | uniq -c > "~{interval_name}_mapq_dist.txt"
+
+    >>>
+
+    runtime {
+        docker: "us.gcr.io/broad-dsde-methods/samtools:v1"
+        disks: "local-disk " + disk_size + " HDD"
+        cpu: cpu
+        memory: memory + " GB"
+    }
+
+    output {
+        File mapq_dist = "~{interval_name}_mapq_dist.txt"
+    }
+}
+
+task CollectMAPQResults {
+    input {
+        String input_name
+        String experiment
+        Array[File] mapq_files
+
+        Int disk_size = 50
+        Int cpu = 2
+        Int memory = 8
+    }
+
+    command <<<
+        python3 << CODE
+
+        import os
+        import pandas as pd
+
+        file_paths = ["~{default="" sep="\", \"" mapq_files}"]
+        full_mapq_summary = pd.DataFrame()
+
+        for file in file_paths:
+            interval_name = os.path.basename(file).split('_mapq_dist.txt')[0]
+
+            mapq_summary = pd.read_csv(file, sep='\s+', names=['Count', 'MAPQ'])
+            mapq_summary['Interval_List'] = interval_name
+            mapq_summary['Sample'] = "~{input_name}"
+            mapq_summary['Experiment'] = "~{experiment}"
+            full_mapq_summary = pd.concat([full_mapq_summary, mapq_summary])
```

Comment:
arent you concatenating an empty dataframe (full_mapq_summary)?

Reply:
For the first iteration, yes, but then the result of the concat gets assigned back to `full_mapq_summary`, so they stack over the full loop.